src/execution_plan/optimizations/optimize_cartesian_product.c

```diff
@@ -99,9 +103,11 @@ static void _optimize_cartesian_product(ExecutionPlan *plan, OpBase *cp) {
 			ExecutionPlan_AddOp(cp, filtered_cp_branch);
 			// Streams are no longer valid since cartesian product changed.
 			for(int j = 0; j < stream_count; j ++) raxFree(stream_entities[j]);
```

Comment:
This should be in the conditional block, as otherwise we'll double free at line 114.